beacon-chain/state/stateutil/trie_helpers.go

```diff
@@ -250,11 +250,11 @@ func AddInMixin(root [32]byte, length uint64) ([32]byte, error) {
 
 // Merkleize 32-byte leaves into a Merkle trie for its adequate depth, returning
 // the resulting layers of the trie based on the appropriate depth. This function
-// pads the leaves to a length of 32.
+// pads the leaves to a length of a multiple of 32.
 func Merkleize(leaves [][]byte) [][][]byte {
 	hashFunc := hash.CustomSHA256Hasher()
 	layers := make([][][]byte, ssz.Depth(uint64(len(leaves)))+1)
-	for len(leaves) != 32 {
+	for len(leaves)%32 != 0 {
```

Comment:
This was a bug where we would allocate infinite memory if the beacon state had more than 32 fields. Wow!

Reply:
bug of the week